### ISSUE TITLE
Async connection manager does not allow you to re-use http-client

### ISSUE THREAD

**Issue by nddipiazza** [dakrone/clj-http#457]
In your documentation, you state that we can reuse `http-client` objects along with `make-reuseable-async-conn-manager`. 

For example:

```
;; Async http clients may also be created and re-used:
(let [acm (conn/make-reuseable-async-conn-manager {})
      ahclient (core/build-async-http-client {} acm "https://example.com" false)]
  (client/get "http://example.com/1"
              {:connection-manager cm :http-client ahclient}
              handle-response handle-failure)
...
```

But when you call `client/get` you will eventually get hit with this code: https://github.com/dakrone/clj-http/blob/3.x/src/clj_http/core.clj#L632

```
    (if-not async?
       (let [^CloseableHttpClient client
             (or http-client
                 (build-http-client req cache?
                                    conn-mgr http-url proxy-ignore-hosts))]
         (try
           (build-response-map (.execute client http-req context)
                               req http-req http-url conn-mgr context client)
           (catch Throwable t
             (when-not (conn/reusable? conn-mgr)
               (conn/shutdown-manager conn-mgr))
             (throw t))))
       (let [^CloseableHttpAsyncClient client
             (build-async-http-client req conn-mgr http-url proxy-ignore-hosts)]
```

When async, this will ignore the `http-client` value you sent in to `client/get` completely and will build a new one every time.

So this will basically rely on you using the async connection pool to avoid building a new http client every time you make a request.

Is this intentional or is this a bug?

**Comment by ricardoekm** [comment]
Looks like a bug. This will force http client to create threads on every request, eliminating the performance advantage of using async requests.

**Comment by rymndhng** [comment]
Fixed in #599
